src/hash_types.rs

```diff
@@ -18,6 +18,12 @@
 
 use hashes::{Hash, sha256, sha256d, hash160};
 
+/// Bitcoin genesis block hash represented as a 32-byte slice in big endian order
+pub const BITCOIN_GENESIS_BLOCKHASH: [u8; 32] = [
```

Comment:
This either belongs in the test module where it is used or into `bitcoin::network::constants`, ideally adding a function to `bitcoin::network::constants::Network` that returns the hash for the respective network (which means adding constants for testnet etc. too).

Reply:
Well, this is a constant, never changes, and it will be strange to call a function just to get a constant value back...

Reply:
Well, it's constant for each chain, but we also want to support testnet etc. and this would make it much easier to write code that works for all networks.

Reply:
This module contains type declaration of other types, so the constant does not belong here. It would be intuitive to look into `hash_types` searching for a constant. 

We already have a function here:

https://github.com/rust-bitcoin/rust-bitcoin/blob/2414c5b0a9bfeccb096ead3596b437dd7b8a8e5e/src/blockdata/constants.rs#L100

If the objective is to save a sha256d computation, I am not opposed to adding another method to it. I also agree with Sebastian that a method is probably the better way to do things rather than a constant.

Reply:
Agreed, I don't care about where the functionality lives as much as that it works with different networks. To build mainnet-only software is an antipattern we shouldn't encourage, testability is important.